api/tx_args.go

```diff
@@ -461,6 +532,302 @@ func (args *AccountUpdateTxArgs) toTransaction() (*types.Transaction, error) {
+		gas = uint64(math.MaxUint64 / 2)
+	}
+	if args.Gas != nil {
+		gas = uint64(*args.Gas)
+	}
+	if globalGasCap != 0 && globalGasCap < gas {
+		logger.Warn("Caller gas above allowance, capping", "requested", gas, "cap", globalGasCap)
+		gas = globalGasCap
+	}
+
+	// Do not update gasPrice unless any of args.GasPrice and args.MaxFeePerGas is specified.
+	gasPrice := new(big.Int)
+	if baseFee.Cmp(new(big.Int).SetUint64(params.ZeroBaseFee)) == 0 {
+		// If there's no basefee, then it must be a non-1559 execution
+		if args.GasPrice != nil {
+			gasPrice = args.GasPrice.ToInt()
+		} else if args.MaxFeePerGas != nil {
+			gasPrice = args.MaxFeePerGas.ToInt()
+		}
+	} else {
+		if args.GasPrice != nil {
+			gasPrice = args.GasPrice.ToInt()
+		} else if args.MaxFeePerGas != nil {
+			// User specified 1559 gas fields (or none), use those
+			gasPrice = args.MaxFeePerGas.ToInt()
+		} else {
+			// User specified neither GasPrice nor MaxFeePerGas, use baseFee
+			gasPrice = new(big.Int).Mul(baseFee, common.Big2)
+		}
+	}
+
+	value := new(big.Int)
+	if args.Value != nil {
+		value = args.Value.ToInt()
+	}
+	data := args.data()
+
+	// TODO-Klaytn: Klaytn does not support accessList yet.
+	// var accessList types.AccessList
+	// if args.AccessList != nil {
+	//	 accessList = *args.AccessList
+	// }
+	return types.NewMessage(addr, args.To, 0, value, gas, gasPrice, data, false, intrinsicGas), nil
+}
+
+// toTransaction converts the arguments to a transaction.
+// This assumes that setDefaults has been called.
+func (args *EthTransactionArgs) toTransaction() (*types.Transaction, error) {
+	var tx *types.Transaction
+	switch {
+	case args.MaxFeePerGas != nil:
+		al := types.AccessList{}
+		if args.AccessList != nil {
+			al = *args.AccessList
+		}
+		tx = types.NewTx(&types.TxInternalDataEthereumDynamicFee{
+			ChainID:      (*big.Int)(args.ChainID),
+			AccountNonce: uint64(*args.Nonce),
+			GasTipCap:    (*big.Int)(args.MaxPriorityFeePerGas),
+			GasFeeCap:    (*big.Int)(args.MaxFeePerGas),
+			GasLimit:     uint64(*args.Gas),
+			Recipient:    args.To,
+			Amount:       (*big.Int)(args.Value),
+			Payload:      args.data(),
+			AccessList:   al,
+		})
+	case args.AccessList != nil:
+		tx = types.NewTx(&types.TxInternalDataEthereumAccessList{
+			ChainID:      (*big.Int)(args.ChainID),
+			AccountNonce: uint64(*args.Nonce),
+			Recipient:    args.To,
+			GasLimit:     uint64(*args.Gas),
+			Price:        (*big.Int)(args.GasPrice),
+			Amount:       (*big.Int)(args.Value),
+			Payload:      args.data(),
+			AccessList:   *args.AccessList,
+		})
+	default:
+		tx = types.NewTx(&types.TxInternalDataLegacy{
+			AccountNonce: uint64(*args.Nonce),
+			Price:        (*big.Int)(args.GasPrice),
+			GasLimit:     uint64(*args.Gas),
+			Recipient:    args.To,
+			Amount:       (*big.Int)(args.Value),
+			Payload:      args.data(),
+		})
+	}
+	return tx, nil
+}
+
+func (args *EthTransactionArgs) ToTransaction() (*types.Transaction, error) {
+	return args.toTransaction()
+}
```

Comment:
it seems that this method is not used anywhere

Reply:
resolved by [6852227](https://github.com/klaytn/klaytn/pull/1894/commits/685222781d7c67f9b055766b948b73e1ca935919)